### PR TITLE
Canvas: Connections positioning ux improvements

### LINKED ISSUE #61423: Canvas: Improve connection positioning UX on element drag / panel resize
Improve connection positioning UX on element drag / panel resize. Ideally the connection(s) should update on every drag event and the connections should be correctly positioned / update on panel resize

https://user-images.githubusercontent.com/22381771/212149338-4f64f443-9be8-4a23-8193-59f5a33708b2.mov

### PR DESCRIPTION
**Pull request by adela-almasan**
Before:

https://user-images.githubusercontent.com/88068998/215580793-9493aa70-2ae1-4c50-ae6e-72360566d56e.mov

After:

https://user-images.githubusercontent.com/88068998/215580290-43133881-ff9b-43d8-b286-51d0d92f9fc4.mov

Fixes #61423

### CODE REVIEW

File: public/app/features/canvas/runtime/scene.tsx [hunk 1]
```diff
@@ -356,6 +359,10 @@ export class Scene {
     return targetElements;
   };
 
+  connectionsNeedUpdate = (element: ElementState): boolean => {
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/scene.tsx:362]: Could this potentially live in the `connections` class?

File: public/app/plugins/panel/canvas/types.ts [hunk 2]
```diff
@@ -31,3 +32,9 @@ export interface CanvasTooltipPayload {
   element: ElementState | undefined;
   isOpen?: boolean;
 }
+
+export interface ConnectionInfo {
```
Comment (nmarrs) [inline public/app/plugins/panel/canvas/types.ts:36]: Nice!

**Review by nmarrs — APPROVED** [review]
This is working really well for me locally, great job! 🎉  Found [this unrelated bug](https://github.com/grafana/grafana/issues/62519) while testing locally

File: public/app/features/canvas/runtime/scene.tsx [hunk 3]
```diff
@@ -78,6 +79,8 @@ export class Scene {
   tooltipCallback?: (tooltip: CanvasTooltipPayload | undefined) => void;
   tooltip?: CanvasTooltipPayload;
 
+  moveableActionCallback?: (moved: boolean) => void;
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/scene.tsx:82]: nice use of a callback :)

File: public/app/features/canvas/runtime/scene.tsx (hunk 1 above)
Comment (adela-almasan) [reply public/app/features/canvas/runtime/scene.tsx:362]: yes! moved it!

**Review by nmarrs — APPROVED** [review]
Refactor changes look good :shipit:

**Pull request merged**